File: src/main/java/it/manuelgozzi/blog/configuration/ManuelGozziBlogConfiguration.java
package it.manuelgozzi.blog.configuration;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

/**
 * @author Manuel Gozzi
 */
@Data
@Configuration
@ConfigurationProperties(prefix = "app")
public class ManuelGozziBlogConfiguration {

    private Auth auth;

    @Data
    public static class Auth {

        private BasicAuthorization basic;
    }

    @Data
    public static class BasicAuthorization {

        private String username;
        private String password;
    }
}

File: src/main/kotlin/it/manuelgozzi/blog/configuration/ManuelGozziBlogConfiguration.kt
package it.manuelgozzi.blog.configuration

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import org.springframework.stereotype.Component

/**
 * @author Manuel Gozzi
 */
@Component
@ConfigurationProperties(prefix = "app")
data class ManuelGozziBlogConfiguration(val auth: Auth = Auth()) {

    data class Auth(var basic: BasicAuthorization = BasicAuthorization())

    data class BasicAuthorization(var username: String? = null, var password: String? = null)
}